lib/vfscore/README.md

```diff
@@ -0,0 +1,131 @@
+The `vnops` structure can be seen as the `file_operation` structure in the Linux Kernel (more as an idea).
+More about this structure [here](https://tldp.org/LDP/lkmpg/2.4/html/c577.htm).
+
+The filesystem library will define two such structures through which it will provide the specified operations.
+To understand how these operations end up being used let's examine the open system call:
+
+```c
+int
+sys_open(char *path, int flags, mode_t mode, struct vfscore_file **fpp)
+{
+        struct vfscore_file *fp;
+        struct vnode *vp;
+        ...
+        error = VOP_OPEN(vp, fp);
+}
+```
+
+`VOP_OPEN()` is a macro that is defined as follows:
+
+```c
+#define VOP_OPEN(VP, FP)           ((VP)->v_op->vop_open)(FP)
+```
+
+## Register a new filesystem
+
+
+To map the desired function for your filesystem with the vfscore application, you need to have declare a `struct vnops` and populate the members, as shown below:
+
+```c
+struct vnops ramfs_vnops = {
+		ramfs_open,             /* open */
+		ramfs_close,            /* close */
+		ramfs_read,             /* read */
+		ramfs_write,            /* write */
+		ramfs_seek,             /* seek */
+		ramfs_ioctl,            /* ioctl */
+		ramfs_fsync,            /* fsync */
+		ramfs_readdir,          /* readdir */
+		ramfs_lookup,           /* lookup */
+		ramfs_create,           /* create */
+		ramfs_remove,           /* remove */
+		ramfs_rename,           /* remame */
+		ramfs_mkdir,            /* mkdir */
+		ramfs_rmdir,            /* rmdir */
+		ramfs_getattr,          /* getattr */
+		ramfs_setattr,          /* setattr */
+		ramfs_inactive,         /* inactive */
+		ramfs_truncate,         /* truncate */
+		ramfs_link,             /* link */
+		(vnop_cache_t) NULL,    /* arc */
+		ramfs_fallocate,        /* fallocate */
+		ramfs_readlink,         /* read link */
+		ramfs_symlink,          /* symbolic link */
+		ramfs_poll,             /* poll */
+};
+```
+
+If some of the functions are not used or not yet defined, you can use the `vfscore_vop_nullop` and cast it to the appropriate type:
+
+```c
+#define ramfs_open      ((vnop_open_t)vfscore_vop_nullop)
+#define ramfs_close     ((vnop_close_t)vfscore_vop_nullop)
+#define ramfs_seek      ((vnop_seek_t)vfscore_vop_nullop)
+```
+
+Let's see now how to link the "file operations" of a filesystem to the `vfscore` library.
+For this, the library exposes a specific structure named `vfscore_fs_type`:
+
+
+```c
+struct vfscore_fs_type {
+        /* name of file system */
+        const char *vs_name;       
+        /* initialize routine */
+        int (*vs_init)(void);
+        /* pointer to vfs operation */
+        struct vfsops *vs_op;
+};
+```
+
+Notice that this structure contains a pointer to the `vfsops` structure, which in turn contains the `vnops` structure.
+To register a filesystem, the `vfscore` library uses an additional section in the ELF.
+You can inspect the [`extra.ld`](/lib/vfscore/extra.ld) file to see it.
```

Comment:
Add a full link here, to avoid messing up the file renderer when we upload this to the website.